### PR TITLE
[ISSUE #1047] support envoy gateway to get XDS rule

### LINKED ISSUE #1047: 支持envoy 独立网关模式
**What is the feature you want to add?**
北极星通过xds协议下发单个domain下多path的访问方式

**Why do you want to add this feature?**
通过单个域名+服务名称 访问北极星的服务
具体访问如下：
`https://envoy.hostname/{serviceName1}/list`
`https://envoy.hostname/{serviceName2}/list`

**How to implement this feature?**
北极星下发xds部分信息如下
image

### PR DESCRIPTION
**Pull request by chuntaojun**
**Please provide issue(s) of this PR:**
Fixes #1047 

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [x] ApiServer
- [ ] Auth
- [ ] Configuration
- [ ] Naming
- [ ] HealthCheck
- [ ] Metrics
- [ ] Docs
- [ ] Installation
- [ ] Performance and Scalability
- [ ] Test and Release

**Please check any characteristics that apply to this pull request.**

- [ ] Does not have any user-facing changes. This may include API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: apiserver/xdsserverv3/gateway.go [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+		VirtualHosts: hosts,
+	}
+
+	return append(routeConfs, routeConfiguration)
+}
+
+// makeGatewayRoutes 构建用于 envoy_gateway 场景的 route.Route 列表
+// 该场景下主要是针对 path 的规则转发，/serviceA => serviceA
+// 当前只有满足以下条件的路由规则支持转为 envoy_gateway 的 xds
+// require 1: 主调服务为全部命名空间&全部服务
```
Comment (andrewshan) [inline apiserver/xdsserverv3/gateway.go:98]: 这个规则是给网关用的，主调是不是必须是网关的服务？

不然可能会和其他服务之间调用的规则混淆

File: apiserver/xdsserverv3/gateway.go [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+// require 2: 请求标签中必须设置 $path 参数
+func (x *XDSServer) makeGatewayRoutes(namespace string) []*route.Route {
```
Comment (andrewshan) [inline apiserver/xdsserverv3/gateway.go:100]: 另外，网关的规则，是不是不能支持被调是统配的场景？

Comment (chuntaojun) [reply apiserver/xdsserverv3/gateway.go:100]: 是的，这里漏了，被调的服务信息必须是精确的

**Review by andrewshan — APPROVED** [review]
LGTM

**Pull request merged**